Generic struct does not implement non-generic interface

<!-- Please make sure to run `v up` before reporting any issues as it may have already been fixed.
     It's also advisable to update all relevant modules using `v outdated` and `v install` -->

<!-- You can use `v doctor` to fill up the next fields -->
**V version:** V 0.2.2 405ed58
**OS:** linux

**What did you do?**
```v
interface Box {
	transform(input int) int
}

struct Test<T> {
	data T
	salt int
}

fn (t Test<T>) transform(input int) int {
	return input + t.salt
}

fn box_transform(b Box) int {
	return b.transform(100)
}

dump(box_transform(Test<string>{
	data: 'hello',
	salt: 6,
}))
```


**What did you expect to see?**
Successfully compile / run and output 106.

 
**What did you see instead?**
C error:

```
==================
/tmp/v/test.14396288592564335338.tmp.c:1805: warning: implicit declaration of function 'main__Test_T_string_transform'
tcc: error: undefined symbol 'main__Test_T_string_transform'
...
==================
```

--- a/vlib/v/checker/checker.v
+++ b/vlib/v/checker/checker.v
@@ -845,6 +845,10 @@ pub fn (mut c Checker) generic_insts_to_concrete() {
 						typ.is_public = true
 						typ.kind = parent.kind
 					}
+					parent_sym := c.table.get_type_symbol(parent_info.parent_type)
+					for method in parent_sym.methods {
+						c.table.register_fn_concrete_types(method.name, info.concrete_types)
+					}
 				}
 				ast.Interface {
 					mut parent_info := parent.info as ast.Interface

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -6652,6 +6652,12 @@ static inline $interface_name I_${cctype}_to_Interface_${interface_name}($cctype
 					continue
 				}
 				// .speak = Cat_speak
+				if st_sym.info is ast.Struct {
+					if st_sym.info.parent_type.has_flag(.generic) {
+						name = g.generic_fn_name(st_sym.info.concrete_types, method.name,
+							false)
+					}
+				}
 				mut method_call := '${cctype}_$name'
 				if !method.params[0].typ.is_ptr() {
 					// inline void Cat_speak_Interface_Animal_method_wrapper(Cat c) { return Cat_speak(*c); }

--- a/vlib/v/tests/generics_struct_with_non_generic_interface_test.v
+++ b/vlib/v/tests/generics_struct_with_non_generic_interface_test.v
@@ -1,0 +1,25 @@
+interface Box {
+	transform(input int) int
+}
+
+struct Test<T> {
+	data T
+	salt int
+}
+
+fn (t Test<T>) transform(input int) int {
+	return input + t.salt
+}
+
+fn box_transform(b Box) int {
+	return b.transform(100)
+}
+
+fn test_generic_struct_with_non_generic_interface() {
+	ret := box_transform(Test<string>{
+		data: 'hello'
+		salt: 6
+	})
+	println(ret)
+	assert ret == 106
+}

